AnnService/inc/Core/Common/FineGrainedLock.h

```diff
@@ -16,43 +16,27 @@ namespace SPTAG
         class FineGrainedLock {
```

Comment:
I didn't check this class usage, but I think we do not allow a mutex object refer twice?
how can you resolve the issue that two threads refer the same mutex when index hash confliction?

Reply:
If two threads try to refer the same mutex and try to lock it with lock_guard, one will wait until the other thread releases the lock.